server/exercise/src/main/protocolBuffers/deviceSensorData.proto

```diff
@@ -0,0 +1,28 @@
+//Scala code generated using https://github.com/SandroGrzicic/ScalaBuff/wiki/Generated-code: sbt run --proto_path=./ --scala_out=./
+//Swift code generated using https://github.com/alexeyxo/protobuf-swift: protoc sensorData.proto --swift_out=./
+
+message MultipleDeviceSensorData {
+  enum Source {
+    HEAD = 0;
+    NECK = 1;
+    SHOULDER = 2;
+    CHEST = 3;
+    ARM = 4;
+    FOREARM = 5;
+    HAND = 6;
+    WRIST = 7;
+    WAIST = 8;
+    HIP = 9;
+    THIGH = 10;
+    CALF = 11;
+    ANKLE = 12;
+    FOOT = 13;
+  }
+
+  message SingleDeviceSensorData {
+    required Source source = 1;
+    required bytes data = 2;
+  }
+
+  repeated SingleDeviceSensorData singleDeviceSensorData = 1;
```

Comment:
Protocol buffers handle this for us. Bytes field is supposed to handle variable length byte array. From what I understand protobuf stores something like [fieldNr][length][data]. So I assume it has to know the length and then uses it when parsing.

We were considering similar functionality if we didn't use protobuf, but we would have to implement it ourselves.